presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1214,13 +1222,14 @@ private void analyzeWindowFunctions(QuerySpecification node, List<Expression> ou
                     throw new SemanticException(NOT_SUPPORTED, node, "FILTER is not yet supported for window functions");
                 }
 
-                Window window = windowFunction.getWindow().get();
+                WindowSpecification windowSpecification = resolveWindowSpecification(windowFunction.getWindow().get(), scope::getWindowSpecification);
+                analysis.setWindowSpecification(windowFunction, windowSpecification);
 
                 ImmutableList.Builder<Node> toExtract = ImmutableList.builder();
                 toExtract.addAll(windowFunction.getArguments());
-                toExtract.addAll(window.getPartitionBy());
-                window.getOrderBy().ifPresent(orderBy -> toExtract.addAll(orderBy.getSortItems()));
-                window.getFrame().ifPresent(toExtract::add);
+                toExtract.addAll(windowSpecification.getPartitionBy());
+                toExtract.addAll(windowSpecification.getOrderBy());
```

Comment:
It's actually the same. `getOrderBy()` returns a list of `SortItem`s.  The difference you see is because, earlier, `Window` had a member `orderBy` of type `Optional<OrderBy>`, but with this change,`orderBy` member in `WindowSpecification` is of type `List<SortItem>`.  Since it's no more Optional, we don't need to include `ifPresent`.